### ISSUE TITLE
How to disable keyring module that was introduced in latest tbc update?

### ISSUE THREAD

**Issue by KonstantinosKoubetsos** [AdiAddons/AdiBags#587]
Also how to remove keyring free space from showing near bag freed space?

**Comment by Talyrius** [comment]
Click the section visibility [V] button at the top-right of the bag to toggle the Key/Keyring sections off.

**Comment by KonstantinosKoubetsos** [comment]
I did that, but the in free space it still shows the free space of the keyring and its annoying.
https://prnt.sc/15sm7pu

**Comment by GregoryGH** [comment]
Yea, same problem. Disabled Keyring but still free space shown. Can you disable it as well? Useless information.

**Comment by gelsbern** [comment]
I reverted back to the previouis version.  The implementation of the keyring messes things up.  Should open it's own window maybe

**Comment by Talyrius** [comment]
**@gelsbern** 
Messes what things up? Be specific.

**Comment by ctthomas** [comment]
![image](https://user-images.githubusercontent.com/7934145/122597180-07f2d480-d039-11eb-899e-f320883e3c85.png)
@Talyrius 
Keyring space is still included in Free Space, even when visibility is unchecked.

**Comment by Talyrius** [comment]
**@ctthomas** 
All of that wasted space is precisely why this option exists:
![image](https://user-images.githubusercontent.com/1068810/122598231-6a3fdb00-d021-11eb-9e60-19b9835f9572.png)

Result:
![image](https://user-images.githubusercontent.com/1068810/122598681-04078800-d022-11eb-93c6-009a94ea6fd3.png)

**Comment by KonstantinosKoubetsos** [comment]
Yes that's true, but still, the merged space showing Keyring free slots is kinda annoying. Not a huge issue, but i would rather not have it. Hence downgrading on earlier version.

**Comment by Talyrius** [comment]
I was told how "annoying" it was that there was no keyring support (see #586) and now I'm hearing about how "annoying" it is now that it's been implemented... There's truly no winning here.

**Comment by ClawhammerLobotomy** [comment]
@Talyrius I'd say the win here would be to have the empty key spaces hide when the key category is hidden. Or have a separate option to show/hide empty key spaces.

Blizzard's UI doesn't even expose the empty key ring spaces to the user since you're likely not going to be filling them up.

I find it helpful to be able to check my keys now, but there isn't much benefit to seeing a number of rarely used empty spaces.

**Comment by Talyrius** [comment]
This is working within the confines of the way AdiBags was written (reusing code for ammo/tradeskill bag support and the like). What you're proposing is a vast departure from that and would require lots of retooling. Sorry, but that's out of the question.

**Comment by KonstantinosKoubetsos** [comment]
Yeah that other request was probably done by people who dont know how their UI works. With bartender / dominos and even classic blizzard UI you can have separate icon for keyring. Clicking that button would open up a bag with their keyring. Am i missing something here? Why i could access it but they couldn't?

**Comment by Talyrius** [comment]
The vast majority of custom UIs do not include bag/keyring buttons.

**Comment by KonstantinosKoubetsos** [comment]
Eh, you are the programmer you make the decisions. But having a cluttered bag for a feature that no one checks is not useful for 3 guys that complained.

**Comment by illutian** [comment]
Would it be possible to have an option to merge free KEY RING space only; separate from the 'normal inventory' merge.

In fact, I'm not sure if I should make this a new post or not, but I imagine the 'specialty bags' (ammo, soul shard, mining, herbalism, etc) could also benefit from being able to merge the free space of just those types. While leaving the omni-purpose free inventory slots alone.

**Comment by KonstantinosKoubetsos** [comment]
Lock specialty bags have feature. Free slots for soul bags is different from normal

**Comment by KonstantinosKoubetsos** [comment]
Anyway i don't want to stress you about this. If you can fix it its ok

**Comment by Psanders0** [comment]
Thank you for all your work on Adibags Talyrius, I think ideally it would be an option here, but looking at the code it is non-trivial to change the bag sources dynamically. 

![image](https://user-images.githubusercontent.com/413803/122627315-a3596900-d07c-11eb-8418-b872e5fb75f3.png)

But in the meantime, for those of us who hate it, there is a small code change that can be made to disable it. (At your own risk) 

Go to \World of Warcraft_classic_\Interface\AddOns\AdiBags\core
Open Constants.lua
Change Line 36 from:
local BAGS = { [KEYRING_CONTAINER] = KEYRING_CONTAINER, [BACKPACK_CONTAINER] = BACKPACK_CONTAINER }
to:
local BAGS = {[BACKPACK_CONTAINER] = BACKPACK_CONTAINER }
Save it
Restart wow

**Comment by mnejing** [comment]
@Talyrius I think the glaring reason people find the update to be annoying is because suddenly we're seeing all of our keys in our bags too, when that wasn't the case prior to like... yesterday. It was jarring to see, and just made my bags look fuller than expected, and honestly kind of made it harder to find items in my bag. 

I realize you're just reusing code (ammo bags etc) to accomplish this, but I expect the more prudent change would be to have a keyring button that opens another window. I STRONGLY suspect that's why people don't like this new thing, because the reality is that the keyring is wholly unimportant to see unless you NEED to see it. There's no need to always know which keys you have, which is the issue right now. 

I'm personally fine with disabling key visibility and seeing the merged keyring free space as a slot in my bag, as it's very minimally intrusive, but that's a personal preference. 

Long story short, the appropriate fix for this is going to require effort. The keyring should be a separate window, not part of your regular bags. Honestly, it shouldn't even be that hard to do....

**Comment by illutian** [comment]
For me, I just like seeing 'at a glance' how many slots I have. Now with the change I needed to use the 'merge' feature and it's a lot more difficult to do a quick check.

I'd prefer the option to not have AdiBags touch the keyring "bag".

**Comment by GregoryGH** [comment]
Just make free keyring slots hidden when keyring itself is hidden and it will be good enough.

**Comment by bidaumm** [comment]
> I was told how "annoying" it was that there was no keyring support (see #586) and now I'm hearing about how "annoying" it is that it's been implemented... There's truly no winning here.

The issue isn't that there is now keyring support. The issue is that the addon by default shows free slots in the keyring in the "free" slot category. This is near irrelevant information to the player and now clutters the addon with unnecessary information.

**Comment by gelsbern** [comment]
To the developers, don't be disheartened, it is great that you are working to try to make everyone happy.  I think what people may have really wanted is to have the keyring as it's own separate display and not merged in to the bags itself.  So Bank, Bags and Keyring all have their own window.  The Blizzard Keyring UI sux and is not moveable, so if the keyring were it's own separate window, I think that would fit  the bill of what I know I was looking for.  I love your addon, keep up the great work and I know you will get this figured out.

**Comment by Zyin055** [comment]
It's great that keyring support was added. But since the keyring is rarely used it clutters the window during normal use. I don't know how the addon is coded, but I feel the best user experience would be to have an option to hide the empty keyring slots, as well as a button on the main window to toggle the occupied keyring slots on/off. This already exists with the "V" button, but it doesn't hide the empty keyring slots.
If that isn't feasible, alternatively you could provide an option to turn off keyring support which could then be re-enabled when the user wants to access the keyring.

**Comment by Talyrius** [comment]
If you're familiar with Lua and want to offer assistance, please see #590.

**Comment by Mrbahaga** [comment]
i uncheck key ring and it didnt remove it from the free space . i had to start using bagnon . even though i dont want to .. till this is fixxed :(

**Comment by jcandresen** [comment]
i don't know what other people are complaining about. i find it super useful to see that I have 32 free slots left on my level 5 druid /s
![image](https://user-images.githubusercontent.com/23084292/124364587-b84b0600-dbff-11eb-904c-16ce95282b9d.png)

**Comment by Calzeru** [comment]
Would love to see this keyring free slots go away. Like someone suggested if you disable keyring then keyring free slots should also disappear.

**Comment by alpturedi** [comment]
> Thank you for all your work on Adibags Talyrius, I think ideally it would be an option here, but looking at the code it is non-trivial to change the bag sources dynamically.
> 
> ![image](https://user-images.githubusercontent.com/413803/122627315-a3596900-d07c-11eb-8418-b872e5fb75f3.png)
> 
> But in the meantime, for those of us who hate it, there is a small code change that can be made to disable it. (At your own risk)
> 
> Go to \World of Warcraft_classic_\Interface\AddOns\AdiBags\core
> Open Constants.lua
> Change Line 36 from:
> local BAGS = { [KEYRING_CONTAINER] = KEYRING_CONTAINER, [BACKPACK_CONTAINER] = BACKPACK_CONTAINER }
> to:
> local BAGS = {[BACKPACK_CONTAINER] = BACKPACK_CONTAINER }
> Save it
> Restart wow

For those who are downgrading, this a pretty awesome solution too.

**Comment by GregoryGH** [comment]
Is there any improvements planned or I should remove it after every update myself?

**Comment by pbjtime** [comment]
lmao what is going on delete this trash feature
